app/models/user.rb

```diff
@@ -50,7 +50,7 @@ def self.find_for_google_oauth2(access_token, signed_in_resource=nil)
     email = data['email'].downcase
     user = User.where(:email => email).first
 
-    raise "This user is hidden!" if user && user.hidden?
+    raise "This user is hidden!" if user && user.hidden_at.present?
```

Comment:
May I suggest:
```
raise UserHiddenError.new if user&.hidden_at
```
I don't think `present?` is necessary, although I could be wrong.